core/src/main/kotlin/api/OrganizationsRoute.kt

```diff
@@ -319,5 +325,65 @@ fun Route.organizations() = route("organizations") {
                 }
             }
         }
+
+        route("groups") {
+            // Instead of identifying arbitrary groups with a groupId, there are only 3 groups with fixed
+            // groupId "readers", "writers" or "admins".
+            route("{groupId}") {
+                put(putUserToGroup) {
+                    requirePermission(OrganizationPermission.WRITE)
+
+                    val user = call.receive<User>()
+                    val organizationId = call.requireIdParameter("organizationId")
+                    val groupId = call.requireParameter("groupId")
+
+                    try {
+                        organizationService.addUserToGroup(user.username, organizationId, groupId)
+                        call.respond(HttpStatusCode.NoContent)
+                    } catch (e: ResourceNotFoundException) {
```

Comment:
The typical pattern to handle exceptions in the REST API is to map them to HTTP status codes using the _StatusPages_ plugin. Refer to `StatusPages.kt`.

Reply:
That's good. I know something similar in Spring Boot.